docs(subject builder) : docs for the transport and sasl subject builder (#3037)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_snippets/snip-subject-builder-replacement-matchers.adoc

:_mod-docs-content-type: SNIPPET

== Replacement matchers

> The replacement matcher is in this form:

Reviewer: ```suggestion
Replacement matchers control how field values are matched and transformed.
A replacement matcher is defined using this format:
```

[source]
----
/<matching regular expression>/<replacement>/<flags>
----
where:

> * `<matching regular expression>` is the regular expression to test the field value.

Reviewer: ```suggestion
* `<matching regular expression>` is the regular expression used to test the field value.
```

* `<replacement>` defines the replacement value to be used if the regular expression is matched.

> The replacement can refer to groups defined in the regular expression (`$1` for first matching group etc)

Reviewer: ```suggestion
The replacement can refer to groups defined in the regular expression using `$1` for the first matching group and so on.
```
